### PR TITLE
Fixes to user scores API

### PR DESCRIPTION
**Pull request by wilmol**
Add need for authentication to users scores endpoint, also ignore BULK_UPLOAD userId in query

Add new endpoints:
 -  GET     /api/stats/votes/total (total vote count, same as before)
 -   GET     /api/stats/votes/completed (count of votes where completed label = vote label)
 -   GET     /api/stats/votes/completed/ratio (essentially users success rate)

I.e. Encourages people to 'play better'

### CODE REVIEW

**Comment by wilmol** [comment]
@rabidgremlin I thought about making it something like /stats

Then we could have another endpoint for votes for phrases that were marked completed with that label. This would encourage people to 'play better'

**Comment by wilmol** [comment]
``` select userId, count(*) completedVotesMade from votes right join phrases on phrases.phraseId=votes.phraseId where completed = true AND phrases.label=votes.label group by userId order by completedVotesMade desc;```

File: concord-server/server/src/main/java/com/rabidgremlin/concord/resources/StatsResource.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  private static final String USER_TO_IGNORE = "BULK_UPLOAD";
+
+  private final StatsDao statsDao;
+
+  private final Logger log = LoggerFactory.getLogger(StatsResource.class);
+
+  public StatsResource(StatsDao votesDao)
+  {
+    this.statsDao = votesDao;
```
Comment (LaurenceTews) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/resources/StatsResource.java]: votesDao?

File: concord-server/server/src/main/java/com/rabidgremlin/concord/resources/StatsResource.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  @GET
+  @Timed
+  @Path("/completed/ratio")
+  public Response getCompletedRatioOfUserVotesForPhrasesBeyondConsensus(@ApiParam(hidden = true) @Auth Caller caller)
+  {
+    log.info("{} getting ratio of completed user votes for phrases beyond consensus.", caller);
+
+    List<UserVoteCount> completedVoteCounts = withoutIgnoredUsers(statsDao.getCompletedCountOfVotesMadePerUser());
+    List<UserVoteCount> totalVoteCountsForPhrasesPassedConsensus = withoutIgnoredUsers(statsDao.getCountOfVotesMadePerUserForPhrasesBeyondVoteMargin(3));
```
Comment (LaurenceTews) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/resources/StatsResource.java]: hardcoding margin?

Comment (wilmol) [reply concord-server/server/src/main/java/com/rabidgremlin/concord/resources/StatsResource.java]: fixed

File: concord-server/server/README.md [hunk 3]
```diff
@@ -9,40 +9,40 @@ curl -v -X POST http://127.0.0.1:9000/api/sessions --header "Content-Type: appli
 
 ### Get next phrase to label
 ```
-curl -v -X GET http://127.0.0.1:9000/api/phrases/next --header "Authorization: Bearer XXXXX"
+curl -v -X GET http://127.0.0.1:9000/api/phrases/next --header "Authorization: Bearer XXXXXX"
```
Comment (joedai) [inline concord-server/server/README.md]: Why the extra X?

File: concord-server/server/src/main/java/com/rabidgremlin/concord/api/UserVoteCount.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+  public void setVoteCount(int voteCount)
+  {
+    this.voteCount = voteCount;
+  }
+
+  @Override
+  public String toString()
+  {
+    return "UserVoteCount{" +
```
Comment (joedai) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/api/UserVoteCount.java]: Consider using ToStringBuilder https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/ToStringBuilder.html

File: concord-server/server/README.md (hunk 3 above)
Comment (wilmol) [reply concord-server/server/README.md]: was inconsistent with the comment about replacing 'XXXXXX'

File: concord-server/server/src/main/java/com/rabidgremlin/concord/api/UserVoteCount.java (hunk 4 above)
Comment (wilmol) [reply concord-server/server/src/main/java/com/rabidgremlin/concord/api/UserVoteCount.java]: could.. I generated this class with IntelliJ

**Pull request merged**